File: worms.py
n = int(input())
worms = [int(i) for i in input().split()]
k = int(input())
labels = [int(i) for i in input().split()]
for i in range(1, n):
    worms[i] += worms[i - 1]

for c in labels:
    i, j = 0, n - 1
    # Find answer using binary search
    while j - i > 0:
        # Calculate mid_point
        b_m = i + (j - i + 1) // 2
        # Pick first half or second half
        m = j - 1 if b_m == j else i + 1 if b_m == 1 else b_m
        i, j = (i, m) if worms[m] > c else (m, j)
        # Determine the answer when the interval is small enough
        if j - i == 1:
            i, j = (j, j) if worms[i] < c <= worms[j] else (i, i)
    print(str(i+1))

File: worms.cpp
#include <iostream>

using namespace std;

int main()
{
    int n;
    cin>>n;
    int worms[n];
    int s[n];
    for (int i = 0; i < n; i++) {
        cin>>worms[i];
        if(i == 0)
            s[i] = worms[i];
        else
            s[i] = s[i-1] + worms[i];
    }
    int k;
    cin>>k;
    for (int l = 0; l < k; l++) {
        int c;
        cin >> c;
        int i = 0;
        int j = n-1;
        while(j-i>0){
            int bm = i + (j-i+1) / 2;
            int m;
            if(bm==j)
                m=j-1;
            else if(bm==i)
                m=i+1;
            else
                m=bm;
            if(s[m]>c)
                j = m;
            else
                i = m;
            if(j - i == 1){
                if(s[i] < c && c <= s[j]){
                    i = j;
                }else{
                    j = i;
                }
            }
        }
        cout<<i+1<<endl;
    }
}
